processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -135,12 +178,15 @@ public ExtractionFn getExtractionFn()
 
     byte[] extractionFnBytes = extractionFn == null ? new byte[0] : extractionFn.getCacheKey();
 
+    byte[] orderingBytes = StringUtils.toUtf8(ordering);
+
     ByteBuffer boundCacheBuffer = ByteBuffer.allocate(
-        9
+        10
         + dimensionBytes.length
         + upperBytes.length
         + lowerBytes.length
         + extractionFnBytes.length
+        + orderingBytes.length
     );
     boundCacheBuffer.put(DimFilterUtils.BOUND_CACHE_ID)
                     .put(boundType)
```

Comment:
GetDimensionRangeSet of BoundDimFilter (for #2982) only supports Lexicographical ordering right now, so the if(... || alphaNumeric) { return null;} should be changed to if(... || !ordering.equals(lexicographic))